gdb/riscv-tdep.c

```diff
@@ -75,100 +76,101 @@ struct riscv_frame_cache
+  {RISCV_RA_REGNUM, {"ra", "x1"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_SP_REGNUM, {"sp", "x2"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_GP_REGNUM, {"gp", "x3"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_TP_REGNUM, {"tp", "x4"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_T0_REGNUM, {"t0", "x5"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_T1_REGNUM, {"t1", "x6"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_T2_REGNUM, {"t2", "x7"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_FP_REGNUM, {"s0", "x8", "fp"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_S1_REGNUM, {"s1", "x9"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_A0_REGNUM, {"a0", "x10"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_A1_REGNUM, {"a1", "x11"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_A2_REGNUM, {"a2", "x12"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_A3_REGNUM, {"a3", "x13"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_A4_REGNUM, {"a4", "x14"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_A5_REGNUM, {"a5", "x15"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_A6_REGNUM, {"a6", "x16"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_A7_REGNUM, {"a7", "x17"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_S2_REGNUM, {"s2", "x18"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_S3_REGNUM, {"s3", "x19"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_S4_REGNUM, {"s4", "x20"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_S5_REGNUM, {"s5", "x21"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_S6_REGNUM, {"s6", "x22"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_S7_REGNUM, {"s7", "x23"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_S8_REGNUM, {"s8", "x24"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_S9_REGNUM, {"s9", "x25"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_S10_REGNUM, {"s10", "x26"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_S11_REGNUM, {"s11", "x27"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_T3_REGNUM, {"t3", "x28"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_T4_REGNUM, {"t4", "x29"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_T5_REGNUM, {"t5", "x30"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+  {RISCV_T6_REGNUM, {"t6", "x31"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
+
+  {RISCV_PC_REGNUM, {"pc"}, true, true, "org.gnu.gdb.riscv.cpu", general_reggroup},
```

Comment:
GCC puts the FP registers immediately after the general registers, which means FP register numbers won't match between gcc and gdb unless they are being remapped someplace else.

Reply:
That's interesting. I don't know how that mapping works, but clearly it must. The actual numbers used aren't actually changed in this changed. RISCV_PC_REGNUM has had that value since before I started working on it.

Reply:
Maybe no one has tested gdb on a cpu that has hardware floating point?

Reply:
I test against spike with FPU almost every time I push a debug-related change. Floating support definitely works.